### PR TITLE
Add Loss Functions Folder, KL Divergence Loss

### PR DESCRIPTION
**Pull request by dakshitagrawal97**
This PR adds a `loss_functions` folder inside the `ann` folder.  The KL Divergence loss has been implemented and put into this folder.

Tests for KL Divergence loss function have been added as well.

### CODE REVIEW

**Comment by rcurtin** [comment]
If the `loss_functions` folder is meant to hold output layer types, should we move the other output layers into that folder also?  (like, for instance, `NegativeLogLikelihood<>`)

**Comment by sourabhvarshney111** [comment]
@rcurtin I think we should move the output layers in that also.

**Comment by sourabhvarshney111** [comment]
@dakshitagrawal97 Looks like the PR is towards its completion. Can you complete it please?

**Comment by dakshitagrawal97** [comment]
@sourabhvarshney111 please feel free to take this PR.  I am a bit occupied with other commitments. :)  

Sorry for saying on the IRC channel that I would complete this. :(

**Comment by sourabhvarshney111** [comment]
@dakshitagrawal97 Can you provide me with the sources? I would try to complete it asap.

**Comment by dakshitagrawal97** [comment]
The idea was to shift all the loss functions of the ann to one folder. Right now, the KL divergence has been implemented.  Some more general loss functions like hinge loss can be implemented.

Also, the cross entropy and mean squared error have been implemented.  They have to be shifted to this folder.
